pkg/cloud/aws/services/ec2/securitygroups.go

```diff
@@ -334,34 +334,63 @@ func (s *Service) getSecurityGroupTagParams(name string, role v1alpha1.SecurityG
 }
 
 func ingressRuleToSDKType(i *v1alpha1.IngressRule) *ec2.IpPermission {
-	res := &ec2.IpPermission{
-		IpProtocol: aws.String(string(i.Protocol)),
-		FromPort:   aws.Int64(i.FromPort),
-		ToPort:     aws.Int64(i.ToPort),
+	var res *ec2.IpPermission
+	switch i.Protocol {
+	case v1alpha1.SecurityGroupProtocolTCP,
```

Comment:
Is there some documentation that we can link to from here?

Reply:
I'll see if I can dig some up – the current link I have goes to a multi-megabyte HTML page that makes all my fans spin up to look at.

Reply:
@vincepri would https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_IpPermission.html be sufficient?

Reply:
whoops – that's what I put in below, I'll add it here